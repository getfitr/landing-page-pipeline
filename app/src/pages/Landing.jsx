import React from 'react';
import { Link as UiLink, Text } from '@chakra-ui/layout';
import { Link } from 'react-router-dom';
import LandingLayout from '../components/layouts/LandingLayout';
import Hero from '../components/sections/Hero';

const unsplashJoshGordon = "https://unsplash.com/@joshgordon?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText";
const unsplash = "https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText";

const Landing = () => (
  <LandingLayout>
    <Hero
      title="Get fitter with the GetFitr! app"
      subtitle="Use the GetFitr! app to plan your workouts, run them, and analyze your successes."
      image="/img/josh-gordon-fzHmP6z8OQ4-unsplash.jpg"
      imageCredit={
        <>
          Photo by <UiLink href={unsplashJoshGordon}>Josh Gordon</UiLink> on <UiLink href={unsplash}>Unsplash</UiLink>
        </>
      }
      ctaText="Start and GetFitr!"
      ctaLink="/signup"
      ctaRemark="The GetFitr! service is free of charge."
    >
      In addition, GetFitr! is a community-driven development project. You can participate and contribute to it.
      <Text _hover={{ color: ["highlight.900", "highlight.900", "highlight.900", "highlight.900"] }}>
        <Link to="/project">
          Learn more about the project.
        </Link>
      </Text>
    </Hero>
  </LandingLayout>
);

export default Landing;
