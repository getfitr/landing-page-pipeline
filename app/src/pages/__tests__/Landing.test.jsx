import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Landing from '../Landing';

test("Shows the landing page", () => {
  const component = renderer.create(
    <BrowserRouter>
      <Landing />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
