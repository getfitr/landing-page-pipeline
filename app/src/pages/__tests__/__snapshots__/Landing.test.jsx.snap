// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows the landing page 1`] = `
<div
  className="css-1cqrh7h"
>
  <nav
    className="css-11u51jp"
  >
    <div
      className="css-70qvj9"
    >
      <div
        className="css-ichje4"
      >
        <p
          className="chakra-text css-1hgmzub"
        >
          GetFitr!
        </p>
      </div>
    </div>
    <div
      className="menu-toggler css-dsyim5"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
    </div>
    <div
      className="menu-wrapper css-8olkqp"
    >
      <div
        className="css-prcoui"
      >
        <p
          className="chakra-text css-1nlkja4"
        >
          <a
            className="home"
            href="/"
            onClick={[Function]}
          >
            GetFitr! App
          </a>
        </p>
        <p
          className="chakra-text css-1nlkja4"
        >
          <a
            className="project"
            href="/project"
            onClick={[Function]}
          >
            GetFitr! Project
          </a>
        </p>
        <p
          className="chakra-text css-1g5pnks"
        >
          <a
            className="signup"
            href="/signup"
            onClick={[Function]}
          >
            <button
              className="chakra-button signup-button css-1ub6xfx"
              type="button"
            >
              Sign Up for GetFitr!
            </button>
          </a>
        </p>
      </div>
    </div>
  </nav>
  <div
    className="css-12n6rdc"
  >
    <div
      className="chakra-stack css-1e9pxi2"
    >
      <h1
        className="chakra-heading css-17ayy37"
      >
        Get fitter with the GetFitr! app
      </h1>
      <h2
        className="chakra-heading css-nm68l6"
      >
        Use the GetFitr! app to plan your workouts, run them, and analyze your successes.
      </h2>
      <div
        className="css-1uylscf"
      >
        In addition, GetFitr! is a community-driven development project. You can participate and contribute to it.
        <p
          className="chakra-text css-1qssnyg"
        >
          <a
            href="/project"
            onClick={[Function]}
          >
            Learn more about the project.
          </a>
        </p>
      </div>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="chakra-button css-pimgrw"
          type="button"
        >
          Start and GetFitr!
        </button>
      </a>
      <p
        className="chakra-text css-1e1m9zt"
      >
        The GetFitr! service is free of charge.
      </p>
    </div>
    <div
      className="css-2ao8yz"
    >
      <img
        className="chakra-image__placeholder css-1h4350"
        size="100%"
      />
      <p
        className="chakra-text css-977540"
      >
        Photo by 
        <a
          className="chakra-link css-0"
          href="https://unsplash.com/@joshgordon?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
        >
          Josh Gordon
        </a>
         on 
        <a
          className="chakra-link css-0"
          href="https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
        >
          Unsplash
        </a>
      </p>
    </div>
  </div>
  <div
    className="css-1wcxvl2"
  >
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-1upb9tn"
    />
    <div
      className="css-m9ru54"
    >
      <div
        className="copyright css-0"
      >
        © 2021 Carsten Koch – Chakra UI template from
        <a
          className="chakra-link css-0"
          href="https://raptis.wtf/blog/build-a-landing-page-with-chakra-ui-part-1/"
          target="_blank"
        >
           Jim Raptis
        </a>
        .
      </div>
      <div
        className="menu-wrapper css-g3n2lv"
      >
        <div
          className="css-gl7sfg"
        >
          <p
            className="chakra-text css-74evef"
          >
            <a
              className="about"
              href="/about"
              onClick={[Function]}
            >
              About Us
            </a>
          </p>
          <p
            className="chakra-text css-74evef"
          >
            <a
              className="privacy"
              href="/privacy"
              onClick={[Function]}
            >
              Privacy
            </a>
          </p>
          <p
            className="chakra-text css-15kpuo5"
          >
            <a
              className="impress"
              href="/impress"
              onClick={[Function]}
            >
              Impress
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
